Issue with py-mvdtool  on Piz-Daint: Unable to find the Boost header files.  Please set BOOST_ROOT to the root

While building simulation toolchain on Piz-Daint, I see:

```
==> Installing py-mvdtool
==> Searching for binary cache of py-mvdtool
==> Warning: No Spack mirrors are currently configured
==> No binary for py-mvdtool found: installing from source
==> Cloning git repository: https://github.com/BlueBrain/MVDTool.git at tag v2.3.2
Note: checking out 'ab35d06f89fd303b10e4feb9a2add435ad7f097a'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

==> No checksum needed when fetching with git
==> Already staged py-mvdtool-2.3.2-nsrszffzjv6zonknw3edg4mvn6p7cdxw in /scratch/snx3000/bp000174/hbp-spack-deployments-test/softwares/26-01-2020/sources/spack/var/spack/stage/py-mvdtool-2.3.2-nsrszffzjv6zonknw3edg4mvn6p7cdxw
==> No patches needed for py-mvdtool
==> Building py-mvdtool [PythonPackage]
==> Executing phase: 'build'
==> '/opt/python/2.7.15.7/bin/python2.7' '-s' 'setup.py' '--no-user-cfg' 'build'
running build
running build_ext
-- The C compiler identification is Intel 19.0.0.20181018
-- The CXX compiler identification is Intel 19.0.0.20181018
-- Cray Programming Environment 2.6.1 C
-- Check for working C compiler: /scratch/snx3000/bp000174/hbp-spack-deployments-test/softwares/26-01-2020/sources/spack/lib/spack/env/intel/icc
-- Check for working C compiler: /scratch/snx3000/bp000174/hbp-spack-deployments-test/softwares/26-01-2020/sources/spack/lib/spack/env/intel/icc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Cray Programming Environment 2.6.1 CXX
-- Check for working CXX compiler: /scratch/snx3000/bp000174/hbp-spack-deployments-test/softwares/26-01-2020/sources/spack/lib/spack/env/intel/icpc
-- Check for working CXX compiler: /scratch/snx3000/bp000174/hbp-spack-deployments-test/softwares/26-01-2020/sources/spack/lib/spack/env/intel/icpc -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /apps/dom/UES/jenkins/7.0.UP01/mc/easybuild/software/CMake/3.14.5/share/cmake-3.14/Modules/FindBoost.cmake:2165 (message):
  Unable to find the requested Boost libraries.

  Unable to find the Boost header files.  Please set BOOST_ROOT to the root
  directory containing Boost or BOOST_INCLUDEDIR to the directory containing
  Boost's headers.
Call Stack (most recent call first):
  CMakeLists.txt:48 (find_package)


-- Found PythonInterp: /opt/python/2.7.15.7/bin/python2.7 (found version "2.7.15")
-- Found PythonLibs: /opt/python/2.7.15.7/lib/libpython2.7.so
-- pybind11 v2.3.dev1
-- Performing Test HAS_INTEL_IPO
-- Performing Test HAS_INTEL_IPO - Success
-- LTO enabled
-- Configuring incomplete, errors occurred!
See also "/tmp/bp000174/spack-stage/spack-stage-5n79gZ/MVDTool/build/temp.linux-x86_64-2.7/CMakeFiles/CMakeOutput.log".
Traceback (most recent call last):
  File "setup.py", line 94, in <module>
    zip_safe=False,
  File "/scratch/snx3000/bp000174/hbp-spack-deployments-test/softwares/26-01-2020/install/install/cray-cnl6-haswell/intel-19.0.1.144/py-setuptools-40.4.3-xjz7zr/lib/python2.7/site-packages/setuptools/__init__.py", line 140, in setup
    return distutils.core.setup(**attrs)
  File "/opt/python/2.7.15.7/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/opt/python/2.7.15.7/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/opt/python/2.7.15.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/opt/python/2.7.15.7/lib/python2.7/distutils/command/build.py", line 127, in run
    self.run_command(cmd_name)
  File "/opt/python/2.7.15.7/lib/python2.7/distutils/cmd.py", line 326, in run_command
    self.distribution.run_command(command)
  File "/opt/python/2.7.15.7/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "setup.py", line 35, in run
    self.build_extension(ext)
  File "setup.py", line 68, in build_extension
    ["cmake", ext.sourcedir] + cmake_args, cwd=self.build_temp, env=env
  File "/opt/python/2.7.15.7/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '/tmp/bp000174/spack-stage/spack-stage-5n79gZ/MVDTool', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/tmp/bp000174/spack-stage/spack-stage-5n79gZ/MVDTool/build/lib.linux-x86_64-2.7', '-DBUILD_PYTHON_BINDINGS:BOOL=ON', '-DBUILD_UNIT_TESTS:BOOL=OFF', '-DMVDTool_VERSION=2.3.2', '-DPYTHON_EXECUTABLE=/opt/python/2.7.15.7/bin/python2.7', '-DCMAKE_BUILD_TYPE=Release']' returned non-zero exit status 1
==> Error: ProcessError: Command exited with status 1:
    '/opt/python/2.7.15.7/bin/python2.7' '-s' 'setup.py' '--no-user-cfg' 'build'

1 error found in build log:
     15    -- Check for working CXX compiler: /scratch/snx3000/bp000174/hbp-spack-deployments-test/softwares/26-01-2020/sources/spack/lib/sp
           ack/env/intel/icpc
     16    -- Check for working CXX compiler: /scratch/snx3000/bp000174/hbp-spack-deployments-test/softwares/26-01-2020/sources/spack/lib/sp
           ack/env/intel/icpc -- works
     17    -- Detecting CXX compiler ABI info
     18    -- Detecting CXX compiler ABI info - done
     19    -- Detecting CXX compile features
     20    -- Detecting CXX compile features - done
  >> 21    CMake Error at /apps/dom/UES/jenkins/7.0.UP01/mc/easybuild/software/CMake/3.14.5/share/cmake-3.14/Modules/FindBoost.cmake:2165 (m
           essage):
     22      Unable to find the requested Boost libraries.
     23
     24      Unable to find the Boost header files.  Please set BOOST_ROOT to the root
     25      directory containing Boost or BOOST_INCLUDEDIR to the directory containing
     26      Boost's headers.
     27    Call Stack (most recent call first):
```

cc: @matz-e @ferdonline

--- a/var/spack/repos/builtin/packages/py-mvdtool/package.py
+++ b/var/spack/repos/builtin/packages/py-mvdtool/package.py
@@ -28,6 +28,7 @@ class PyMvdtool(PythonPackage):
     depends_on('cmake@3.2:', type='build')
     depends_on('py-numpy', type='run')
 
+    depends_on('boost')
     depends_on('mpi', when='+mpi')
     depends_on('hdf5+mpi', type=('build', 'run'), when="+mpi")
     depends_on('hdf5~mpi', type=('build', 'run'), when="~mpi")

